### PR TITLE
Add ZTP-GitOps workflow role

### PR DESCRIPTION
**Pull request by sshnaidm**
Add role that using existing ACM installation (see other roles for that) installs and configures ZTP GitOps workflow from scratch, configures repositories and ready for usage.

### CODE REVIEW

**Review by ramperher — COMMENTED** [review]
Hi @sshnaidm, thanks for your contributions to redhatci-ocp collections!

I have some general comments to say here, but firstly, I wanted to comment that we already have some GitOps-ZTP roles in the collections repo. They were included in this PR: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/189. I was wondering if this role is really doing something different compared to what it's already in place in the collections, maybe @nsilla who authored that change could also give his opinion here.

Then, moving to the change itself, I have a couple of observations, in case it's confirmed this role is something really different and that it's really required:

- Firstly, the main playbook seems to be quite long, could it be splitted into different playbooks that are called from the main one?
- I'm not sure if kubernetes.core will work since we're still relying on community.kubernetes. The usage of kubernetes.core is being studied in https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/214
- How is this role tested? How can we confirm it's working fine?

File: roles/ztphub/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+Role Variables
+--------------
+
+* `ztphub_working_dir`: (string) Local working directory path for files.
+* `ztphub_gitops_operator_install_channel`: (string) GitOps (ArgoCD) operator channel version (default is `latest`)
+* `ztphub_talm_operator_install_channel`: (string) TALM operator channel version (default is `stable`)
+* `ztphub_image`: (string) Image for ZTP generator (default is `quay.io/openshift-kni/ztp-site-generator`)
+* `ztphub_out_dir`: (string) Directory path for extracting files from ZTP image (default is `{{ ztphub_working_dir }}/out`)
+* `ztp_automatic_update`: (bool) Whether to update automatically operators (`installPlanApproval`) (default is `true`)
+* `ztphub_install_ops_mode`: (string) `Manual` or `Automatic` depends on `ztp_automatic_update`.
```
Comment (nsilla) [inline roles/ztphub/README.md:24]: I'm not entirley convinced of how the ztphub_automatic_update and ztphub_install_ops_mode pair work together, but regardless of this, note that the name of the ztphub_automatic_update is wrong (it's missing the "hub" part).

**Comment by ramperher** [comment]
> * sure we can split it, and probably to extract gitops operator and talm installs to a different role
:+1: , also, please pay attention to all the checks that are failing and you can see it below (Ansible-lint check, sanity checks, etc.), they're also required before accepting the change.

> * I'm not sure I understand how `kubernetes.core` is blocking this role, you can use whatever version, we don't use obsolete `community.kubernetes`, but only `kubernetes.core`. This patch has nothing to do with your testing of  `kubernetes.core`.
I am saying this because this collection currently requires `community.kubernetes`, but `kubernetes.core` is still not included as requirement, as you can see in the [spec file](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/ansible-collection-redhatci-ocp.spec#L21).

As commented before, in this change, there's an ongoing work to move to kubernetes.core: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/214. What I'm not sure is that there's any implication to use kubernetes.core in a specific role like this one. CC @tonyskapunk here since he was working in the transition towards `kubernetes.core`, just to be sure.

> * We use it in our workflow and working on testing in molecule, probably will be added in a different Github runner, because standard ones can't handle it. I wonder though how and where you test other gitops roles? And all other roles in this repo? DCI testing is not relevant for us.

If this is not intended to be tested with DCI, I'd recommend to include `Test-Hints: no-check` in the PR description, to avoid launching DCI jobs each time there's a new commit in this PR. Something similar to what was done in this PR: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/257

**Comment by tonyskapunk** [comment]
> As commented before, in this change, there's an ongoing work to move to kubernetes.core: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/214. What I'm not sure is that there's any implication to use kubernetes.core in a specific role like this one.

While is not a requirement, it's best to keep cohesiveness with other roles as they are part of a collection that has the dependency on community.kubernetes

https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/4bfd5aa02c5cbc57f4bc95df293d8a50e20577b3/galaxy.yml#L103

**Review by rlmanrique — COMMENTED** [review]
In SLCM we are also [setting allowEmpty](https://github.com/openshift-kni/eco-validation/blob/dev/roles/utils/ztp/configure_ztp_gitops_apps/tasks/main.yaml#L84-L88) and configure the [clusterImage set](https://github.com/openshift-kni/eco-validation/blob/dev/roles/utils/ztp/configure_ztp_gitops_apps/tasks/main.yaml#L91-L100)

Didn't you consider to add those parameters?

File: roles/ztphub/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    - '^\s+selfHeal:.*'
+  when: not ztphub_app_autosync
+
+- name: Deploy application
+  shell: >-
+    oc --kubeconfig={{ ztphub_kubeconfig_file }} apply -k {{ ztphub_out_dir }}/argocd/deployment
+  changed_when: "' changed' in deploy_result.stdout or ' created' in deploy_result.stdout"
+  register: deploy_result
+
+- name: Connect repo by creating secret in ArgoCD
```
[the remark this answers is not in the document]
Comment (rlmanrique) [reply roles/ztphub/tasks/main.yml:127]: For this reason, configurig the repo could be automated in a separated role as mentioned in the meeting.

**Comment by nsilla** [comment]
Hi guys,

This PR is been staled for some time now.

The last week some of the interested people had a meeting to discuss a plan for it, and the idea was to have this change approved, so @sshnaidm could start integrating his code with the collection, while in parallel we'd work in merging this role with the other existing ones for the gitops-ztp workflows so, eventually, this role won't be needed.

If anybody has further concerns I'd like to approve it.

**Comment by tonyskapunk** [comment]
> The last week some of the interested people had a meeting to discuss a plan for it, and the idea was to have this change approved, so @sshnaidm could start integrating his code with the collection, while in parallel we'd work in merging this role with the other existing ones for the gitops-ztp workflows so, eventually, this role won't be needed.

I image that the role will be adjusted to use community.kubernetes? Otherwise I'm not sure how is going to be integrated, this is still pending from https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/262#issuecomment-2061874517

**Pull request closed without merging**